src/test/java/org/openstreetmap/atlas/tags/filters/matcher/parsing/tree/TreePrinterTest.java

```diff
@@ -0,0 +1,63 @@
+package org.openstreetmap.atlas.tags.filters.matcher.parsing.tree;
+
+import org.junit.Assert;
+import org.junit.Test;
+import org.openstreetmap.atlas.tags.filters.matcher.parsing.Lexer;
+import org.openstreetmap.atlas.tags.filters.matcher.parsing.Parser;
+
+/**
+ * @author lcram
+ */
+public class TreePrinterTest
+{
+    @Test
+    public void test()
+    {
+        final String input1 = "foo=bar | !(baz=bat)";
+        ASTNode root = new Parser(new Lexer().lex(input1), input1).parse();
+        final String tree1 = TreePrinter.print(root);
+        Assert.assertEquals(
+                "                                |                               \n"
+                        + "                ┌───────────────┴───────────────┐               \n"
```

Comment:
This is a bold test... What if we change the printing slightly in the future? It has to be all re-printed here?

Reply:
Ha this is a good point. I actually do want to update this in the future to improve the printing of `!` operators. So I will leave these tests in place for now to make sure when I do fiddle with things, it doesn't break.